clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -248,6 +250,9 @@ public void add(MetricName metricName, MeasurableStat stat) {
      * @param config A special configuration for this metric. If null use the sensor default configuration.
      */
     public synchronized void add(MetricName metricName, MeasurableStat stat, MetricConfig config) {
+        if (hasExpired())
+            throw new IllegalStateException("Sensor expired!!");
```

Comment:
Maybe the message can say something like "Cannot add a new metric to an expired sensor"?